execution.go

```diff
@@ -1,133 +1,21 @@
 package dexec
 
-import (
-	"errors"
-	"fmt"
-	"io"
+import "io"
 
-	"github.com/fsouza/go-dockerclient"
-)
+type ContainerClient interface {
+	Docker | Containerd
+}
 
 // Execution determines how the command is going to be executed. Currently
 // the only method is ByCreatingContainer.
-type Execution interface {
-	create(d Docker, cmd []string) error
-	run(d Docker, stdin io.Reader, stdout, stderr io.Writer) error
-	wait(d Docker) (int, error)
+type Execution[T ContainerClient] interface {
+	create(d T, cmd []string) error
```

Comment:
This predates you and the whole thing is one package, but un-exported functions on an interface?  I guess I thought I had seen it all

Reply:
yea, pretty odd but what can ya do